utils/io/src/commonMain/kotlin/Path.kt

```diff
@@ -225,6 +225,8 @@ expect val SystemPath.absolutePath: String
 
 expect inline fun <T> SystemPath.useDirectoryEntries(block: (Sequence<SystemPath>) -> T): T
 
+expect fun SystemPath.moveDirectoryRecursively(target: SystemPath, visitor: ((SystemPath) -> Unit)? = null)
```

Comment:
```suggestion
expect fun SystemPath.moveDirectoryRecursively(target: SystemPath, onMoved: ((SystemPath) -> Unit)? = null)
```

visitor 听起来像是它要实现 move 功能